jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -28,24 +30,72 @@ class Catalogs(PluginTemplateMixin, ViewerSelectMixin):
     template_file = __file__, "catalogs.vue"
     catalog_items = List([]).tag(sync=True)
     catalog_selected = Unicode("").tag(sync=True)
+    from_file = Unicode().tag(sync=True)
+    from_file_message = Unicode().tag(sync=True)
     results_available = Bool(False).tag(sync=True)
     number_of_results = Int(0).tag(sync=True)
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        # TODO: Add more catalogs.
+
         self.catalog = SelectPluginComponent(self,
                                              items='catalog_items',
                                              selected='catalog_selected',
-                                             manual_options=['SDSS'])
+                                             manual_options=['SDSS', 'From File...'])
+
+        # file chooser for From File
+        start_path = os.environ.get('JDAVIZ_START_DIR', os.path.curdir)
+        self._file_upload = FileChooser(start_path)
+        self.components = {'g-file-import': self._file_upload}
+        self._file_upload.observe(self._on_file_path_changed, names='file_path')
+        self._cached_table_from_file = {}
+
+    def _on_file_path_changed(self, event):
+        self.from_file_message = 'Checking if file is valid'
+        path = event['new']
+        if (path is not None
+                and not os.path.exists(path)
+                or not os.path.isfile(path)):
+            self.from_file_message = 'File path does not exist'
+            return
+
+        try:
+            table = QTable.read(path)
+        except Exception:
+            self.from_file_message = 'Could not parse file with astropy.table.QTable.read'
+            return
+
+        if 'sky_centroid' not in table.keys():
+            self.from_file_message = 'Table does not contain required sky_centroid column'
+            return
+
+        # since we loaded the file already to check if its valid, we might as well cache the table
+        # so we don't have to re-load it when clicking search.  We'll only keep the latest entry
+        # though, but store in a dict so we can catch if the file path was changed from the API
+        self._cached_table_from_file = {path: table}
```

Comment:
Is there a way for user to invalidate cache? What if user modified the table on disk after loading it into the plugin?

Reply:
I had considered that... but from a UX perspective I'm not really sure what would be expected in this scenario (the button says "search", not load, so to me it makes sense that the file is already loaded and parsed once selected and so future changes to the file would require re-loading it).  @Jenneh - any thoughts?

Reply:
For some context: I had originally gone against "Load" to avoid any relation to the load data dialog (which has a "Load" button).

Reply:
After discussing with @Jenneh - I'll change the text on the button to "Load" and we can always revisit the behavior if it causes any confusion (but we don't watch for file changes anywhere else in the app - once you load a fits file, for example, changes on disk are not picked up).